include/ecs/game_coordinator.h

```diff
@@ -0,0 +1,83 @@
+#ifndef ECS_GAME_COORDINATOR_H
+#define ECS_GAME_COORDINATOR_H
+
+#define template_base(T, B) template <typename T, std::enable_if_t<std::is_base_of_v<B, T>, bool> = true>

```

Comment:
I think using [C++20 concepts](https://en.cppreference.com/w/cpp/language/constraints) here could turn out to be much better.

Reply:
I'll read into it

Reply:
Done